implementations/rust/ockam/ockam_command/src/vault/list.rs

```diff
@@ -25,11 +29,67 @@ impl ListCommand {
     }
 }
 
+pub struct VaultListOutput {
+    pub name: String,
+    pub is_aws: bool,
+    pub is_default: bool,
+}
+
+impl VaultListOutput {
+    pub fn new(name: String, is_aws: bool, is_default: bool) -> Self {
+        Self {
+            name,
+            is_aws,
+            is_default,
+        }
+    }
+}
+
+impl Output for VaultListOutput {
+    fn output(&self) -> crate::error::Result<String> {
+        let mut output = String::new();
+        writeln!(
+            output,
+            "Vault {} {}",
+            self.name
+                .to_string()
+                .color(OckamColor::PrimaryResource.color()),
+            if self.is_default { "(default)" } else { "" }
+        )?;
+        write!(
+            output,
+            "Type {}",
+            match self.is_aws {
+                true => "AWS KMS",
+                false => "OCKAM",
+            }
+            .to_string()
+            .color(OckamColor::PrimaryResource.color())
+        )?;
+        Ok(output)
+    }
+}
+
 fn run_impl(opts: CommandGlobalOpts) -> miette::Result<()> {
     let vaults = opts.state.vaults.list()?;
-    let plain = opts
-        .terminal
-        .build_list(&vaults, "Vaults", "No vaults found on this system.")?;
+    let mut vault_output: Vec<VaultListOutput> = Vec::new();
+    let mut default_vault_name = String::new();
+    if let Ok(default_vault) = opts.state.vaults.default() {
+        default_vault_name = default_vault.name().to_string();
+    }
+
+    for vault in &vaults {
+        let i = VaultListOutput::new(
+            vault.name().to_string(),
+            vault.is_aws(),
+            default_vault_name == *vault.name(),
+        );
+        vault_output.push(i);
+    }
+
+    let plain =
+        opts.terminal
+            .build_list(&vault_output, "Vaults", "No vaults found on this system.")?;
```

Comment:
You can also simplify a little how you build the output:

```rust
let output = vaults
    .iter()
    .map(|v| {
        VaultListOutput::new(
            v.name().to_string(),
            v.config().is_aws(),
            opts.state.vaults.is_default(v.name()).unwrap_or(false),
        )
    })
    .collect::<Vec<_>>();
```

Now that we have a proper type to handle the vault ouptut, you could use it for all the formats, so that the output has the same structure:

```rust
let plain = opts
    .terminal
    .build_list(&output, "Vaults", "No vaults found on this system.")?;
let json = serde_json::to_string_pretty(&output).into_diagnostic()?;
```

For this to work though, you will need to add `#[derive(serde::Serialize)]` to the `VaultListOutput`.

Reply:
Ahh that is simpler, thanks! 
The current json output looks like this:
```
    Running `/Users/melissali/Documents/ockam/ockam/target/debug/ockam vault list --output json`
[
  {
    "name": "v1",
    "path": "/Users/melissali/.ockam/vaults/v1.json",
    "data_path": "/Users/melissali/.ockam/vaults/data/v1-storage.json",
    "config": {
      "aws_kms": false
    }
  },
  {
    "name": "v2",
    "path": "/Users/melissali/.ockam/vaults/v2.json",
    "data_path": "/Users/melissali/.ockam/vaults/data/v2-storage.json",
    "config": {
      "aws_kms": false
    }
  }
]
```

If I use the vault output for the json output (after I also add more attributes to the VaultListOutput struct), I can get it to look like this:
```
     Running `/Users/melissali/Documents/ockam/ockam/target/debug/ockam vault list --output json`
[
  {
    "name": "v1",
    "config": {
      "aws_kms": false
    },
    "is_default": false,
    "path": "/Users/melissali/.ockam/vaults/v1.json",
    "data_path": "/Users/melissali/.ockam/vaults/data/v1-storage.json"
  },
  {
    "name": "v2",
    "config": {
      "aws_kms": false
    },
    "is_default": true,
    "path": "/Users/melissali/.ockam/vaults/v2.json",
    "data_path": "/Users/melissali/.ockam/vaults/data/v2-storage.json"
  }
]
```
Is this the same idea you had? I tried to preserve all the original keys, just adding the "is_default" field.

Reply:
You can drop the `path` and `data_path` fields, as they are not very useful to the CLI user. Also, you could flatten the `config` field by adding the `#[serde(flatten)]` attribute to that field in the `VaultListOutput` struct.